src/neo/SmartContract/InteropService.Oracle.cs

```diff
@@ -0,0 +1,36 @@
+using Neo.Oracle.Protocols.HTTP;
+using System;
+
+namespace Neo.SmartContract
+{
+    partial class InteropService
+    {
+        public static class Oracle
+        {
+            public static readonly uint System_Oracle_Get = Register("System.Oracle.Get", Oracle_Get, 0, TriggerType.Application, CallFlags.None);
+
+            private static bool Oracle_Get(ApplicationEngine engine)
+            {
+                if (engine.OracleCache == null) return false;
```

Comment:
Is it possible that `engine.OracleCache` is `null`?

Reply:
Yes, with regultar TX is null, so it will fail.
Only wont be null if is the caller is oracle service, or if it's already processed by the oracle and cached.

Reply:
How to distinguish between regultar transactions and oracle transactions?

Reply:
I think we have oracle transaction type as we discussed #1274 , right? @shargon @belane

Reply:
I think that a flag for transaction it's not bad, but it are contradictory with one of the NEO3 features, unique kind of transactions, that's the reason for distinguish with the `TransactionAttributes`

Reply:
No @Tommo-L, we think that the best option is through attributes, it is as we designed at the beginning.
The other option would involve creating a new type of transaction, something we wanted to avoid.

Reply:
Got, I think it's ok.